File: channel/model/channel.go
package model

import (
	"time"
)

type Channel struct {
	ID uint64 `gorm:"primary_key;not null;AUTO_INCREMENT" json:"-" yaml:"-"`

	UUID        string    `gorm:"not null"                    json:"id"          yaml:"id"`
	CreatedAt   time.Time `gorm:"not null"                    json:"createdAt"   yaml:"createdAt"`
	UpdatedAt   time.Time `gorm:"not null"                    json:"updatedAt"   yaml:"updateAt"`
	Title       string    `gorm:"not null"                    json:"title"       yaml:"title"`
	Owner       string    `gorm:"not null"                    json:"owner"       yaml:"owner"`
	Description string    `gorm:"not null"                    json:"description" yaml:"description"`
	Programs    []Program `gorm:"many2many:channel_programs;" json:"programs"    yaml:"programs"`
}

File: console/src/shared/channel.ts
import { Program } from './program';

export class Channel {
    id: string;
    title: string;
    owner: string;
    description: string;
    createdAt: Date;
    updatedAt: Date;
    programs: Array<Program>;

    constructor(id: string, title: string, owner: string, description?: string, programs?: Array<Program>, createdAt?: Date) {
        this.id = id;
        this.title = title;
        this.owner = owner;
        this.description = description || '';
        this.programs = programs || [];
        this.createdAt = createdAt || new Date();
    }
}

export let NullChannel = new Channel(undefined, undefined, undefined);
